placekeeper.js

```diff
@@ -1,1056 +1,1056 @@
+            }
+            if (elem.attachEvent) {
+                return elem.attachEvent("on" + event, function(e) {
+                    e.preventDefault = function() {
+                        e.returnValue = false;
+                    };
+                    e.stopPropagation = function() {
+                        e.cancelBubble = true;
+                    };
+                    fn.call(elem, e);
+                });
+            }
+        }
+
+        function removeEventListener(elem, event, fn) {
+            if (elem.removeEventListener) {
+                return elem.removeEventListener(event, fn, false);
+            }
+            if (elem.detachEvent) {
+                return elem.detachEvent("on" + event, fn);
             }
-            data.setTypeAttr(element, "password");
         }
 
-        element.value = val;
-        data.setActiveAttr(element);
-        utils.addClass(element, "placeholder");
-        storeMaxlength(element);
-    }
+        function trim(str) {
+            return str.replace(/^\s+|\s+$/g, "");
+        }
 
-    function removePlaceholder(element, replace) {
-        element.value = replace ? element.value.replace(data.getValueAttr(element), "") : "";
-        data.removeActiveAttr(element);
-        utils.removeClass(element, "placeholder");
-        restoreMaxlength(element);
-    }
+        function hasClass(elem, className) {
+            return (" " + elem.className + " ").indexOf(" " + className + " ") !== -1;
+        }
 
-    function hidePlaceholder(element) {
+        function addClass(elem, className) {
+            if (!hasClass(elem, className)) {
+                elem.className = elem.className === "" ? className : elem.className + " " + className;
+            }
+        }
 
-        if (!data.hasActiveAttrSetToTrue(element)) {
-            return;
+        function removeClass(elem, className) {
+            elem.className = trim((" " + elem.className + " ").replace(" " + className + " ", " "));
         }
 
-        if (data.hasTypeAttrSetToPassword(element)) {
-            if (elems.isClonedPasswordInput(element)) {
-                var original = elems.getPasswordOriginal(element);
-                elems.swapElements(element, original);
-                element = original;
-                element.focus();
-            } else {
-                element.type = "password";
+        function getAttributes(elem) {
+            var newAttrs = {};
+            var attrs = elem.attributes;
+            for (var i = 0; i < attrs.length; i++) {
+                // old IEs will throw an error if you try to copy "type" attribute.
+                if (attrs[i].specified && attrs[i].name !== "type") {
+                    newAttrs[attrs[i].name] = attrs[i].value;
+                }
             }
+            return newAttrs;
         }
 
-        removePlaceholder(element, true);
-    }
+        function setAttributes(elem, attrs) {
+            for (var key in attrs) {
```

Comment:
The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.